src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaChoiceRestriction.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Validator\Metadata\Property\Restriction;
+
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Choice;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class PropertySchemaChoiceRestriction implements PropertySchemaRestrictionMetadataInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
+    {
+        if (!\in_array($builtinType = $propertyMetadata->getType()->getBuiltinType(), [Type::BUILTIN_TYPE_STRING, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
+            return [];
+        }
+
+        if (\is_callable($choices = $constraint->callback)) {
+            $choices = $choices();
+        } elseif (\is_array($constraint->choices)) {
+            $choices = $constraint->choices;
+        } else {
+            $choices = [];
+        }
+
+        if (!$choices) {
+            return [];
+        }
+
+        if ($constraint->multiple) {
+            $restriction['type'] = 'array';
+            $restriction['items'] = ['type' => Type::BUILTIN_TYPE_STRING !== $builtinType ? 'string' : 'number', 'enum' => $choices];
```

Comment:
Isn't the condition reversed?
```suggestion
            $restriction['items'] = ['type' => Type::BUILTIN_TYPE_STRING === $builtinType ? 'string' : 'number', 'enum' => $choices];
```

Reply:
I think the tests are not working as expected id it's the case.

Reply:
this is by accident, now locally tests was failing for me, will push in a second